grpc/binlog/v1alpha/binarylog.proto

```diff
@@ -31,8 +31,14 @@ message GrpcLogEntry {
     SEND_TRAILING_METADATA = 2;
     SEND_MESSAGE = 3;
     RECV_INITIAL_METADATA = 4;
+    // On server side, this event means the client has half-closed.
+    // On client side, this is means the RPC has now ended.
     RECV_TRAILING_METADATA = 5;
     RECV_MESSAGE = 6;
+    // Client side only.
+    CLIENT_SEND_CANCEL = 7;
```

Comment:
What happens if the call is cancelled from the server side?  Would it make sense to split this up into SEND_CANCEL and RECV_CANCEL, either of which can be used on the client or server side?

Reply:
On client side, inspecting the status would be enough to determine if the server cancelled the call because an RPC always ends with some sort of status (even if locally generated by gRPC).

On server side, applications do not know why a RPC abnormally terminated. It can be that the client sent a cancellation, or something else like deframing failed, a ping timed out, or server shutdown was initiated. These are all conflated together as [onCancel](https://grpc.io/grpc-java/javadoc/io/grpc/ForwardingServerCallListener.html#onCancel--).

Reply:
Again, with directly logging the status, we would not need to log CLIENT_SEND_CANCEL.